prow/plugins/assign/assign.go

```diff
@@ -159,9 +168,12 @@ type handler struct {
 	log *logrus.Entry
 	// userType is a string that represents the type of users affected by this handler. (e.g. 'assignees')
 	userType string
+
+	// Counter is a prometheus counter for tracking plugin usage 
+	counter prometheus.Counter
 }
 
-func newAssignHandler(e github.GenericCommentEvent, gc githubClient, log *logrus.Entry) *handler {
+func newAssignHandler(e github.GenericCommentEvent, gc githubClient, log *logrus.Entry, c prometheus.Counter) *handler {
```

Comment:
Q1: do I need to pass the counter as reference, or `counter.Inc()` knows about the parent CounterVec?
Q2: How do I test this

Reply:
Q1: I think it is fine as is: https://godoc.org/github.com/prometheus/client_golang/prometheus#Counter
Q2: You can implement the `Counter` interface with a test counter and check that it is incremented when expected.